### ISSUE TITLE
[Suggestion] Auto-convert extensions do URI components in project.json for offline use

### ISSUE THREAD

**Issue by IceandFire04** [PenguinMod/PenguinMod-Vm#38]
This is something that would be nice for using PenguinMod at school because most schools block extension.turbowarp.org. This would be another option in the settings tab called "Load extensions as data URIs," which explains that you may not be able to upload the project if you can't upload custom extensions. Also good for offline use.

**Comment by RedMan13** [comment]
skill issue? 💀

**Comment by JeremyGamer13** [comment]
this is planned to be an option upon uploading

**Comment by JeremyGamer13** [comment]
for regular projects im not sure

**Comment by IceandFire04** [comment]
Cool 😎

**Comment by SharkPool-SP** [comment]
> for regular projects im not sure

jsyk its `encodeuricomponent()` not just a data.uri

**Comment by IceandFire04** [comment]
Well I don't know how it works internally so idk
But yeah that's what I mean ig

**Comment by Ianyourgod** [comment]
> skill issue? 💀

on god

**Comment by AD1340** [comment]
i need this same thing brah

**Comment by AD1340** [comment]
when i try to load my project on a school network, i get failed to fetch error but when i use vpn it works

same thing happens when i try to play some projects on the website itself, failed to fetch but when i use vpn on the school wifi it works

**Comment by IceandFire04** [comment]
Yeah schools block turbowaro because to the average district admin it looks like a crappy unblocked games website (not their fault, they have more important things to do)

The reason this is an issue is extensions are typically blocked as well

Right now just download the extensions from GitHub and upload them yourself, but the issue is no uploading the project

**Comment by AD1340** [comment]
this is a very big issue because what if the extension links get taken down
